README.md

```diff
@@ -0,0 +1,219 @@
+    def add_to_cart
+      # Get the item from the path
+      @item = Item.find(params[:id])
+      
+      # Load the cart from the session, or create a new empty cart.
+      cart = session[:cart] || []
+      cart << @item.id
+
+      # Save the cart in the session.
+      session[:cart] = cart
+    end
+
+That's it! It's common to wrap this up in a helper method:
+
+It's common to wrap up some of this functionality in a helper method:
+
+    class ApplicationController < ActionController::Base
+      def current_cart
+        session[:cart] ||= []
+      end
+    end
+
+So now our controller looks like this:
+
+    # Routed from POST /items/:id/add_to_cart
+    def add_to_cart
+      # Get the item from the path
+      @item = Item.find(params[:id])
+      
+      # Load the cart from the session, or create a new empty cart.
+      current_cart << @item.id
+    end
+
+This way, we can use `current_cart` in our views and layouts too. For example, we may want to show the user how many items are in their cart as part of the layout.
+
+## Conclusion
+
+Cookies are foundational for the modern web.
+
+You've probably heard a lot about cookies from non-technical sources. In fact, the EU recently passed a law requiring sites that use cookies to request user consent before doing so.
+
+The effect of this law was not unlike the effect of California's [Proposition 65][prop_65], which states that buildings must have clear signage if they contain any one of a list of chemicals known to the state of California to cause cancer.
+Nobody knows or wants to put in the considerable expense to figure out what's in their building, and signs are basically free, so literally every building in California now carries a prominent sign telling you it causes cancer.
+
+Most sites use cookies, either to let their users log in, or to keep track of their
+shopping carts, or record other ephemeral session data. Almost nobody thinks these are bad uses of cookies: nobody really believes that you should have to type in your username and password on every page, or that your shopping cart should clear if you reload the page.
+
+But cookies just let you store data on in a user's browser, so by nature, they can be used for more controversial endeavors. 
+
+For example, Google AdWords sets a cookie and uses that cookie to track what ads you've seen and which ones you've clicked on. The tracking information helps AdWords decide what ads to show you.
+
+This is why, if you click on an ad, you may find that the ad follows you around the Internet. It turns out that this behavior is as effective as it is annoying: people are far more likely to buy things from ads that they've clicked on once.
+
+This use of cookies worries people. And so now the EU has a law.
+
+Cookies, like any technology, are a tool. In the rest of this unit, we're going to be using them to let users log in. Whether you later want to use them in such a way that the EU passes another law is up to you.
```

Comment:
I think reminding students to be aware of what they are saving and to do right by their users is important.